File: src/main/kotlin/net/squarelabs/pgrepl/services/ConfigService.kt
package net.squarelabs.pgrepl.services

class ConfigService {
    private val dbHost = "localhost"
    private val dbName = "pgrepl_test"

    fun getDbHost(): String {
        return System.getenv("DB_HOST") ?: dbHost
    }

    fun getJdbcDatabaseUrl(): String {
        return "jdbc:postgresql://${getDbHost()}:5432/postgres?user=postgres&password=postgres"
    }

    fun getAppDbName(): String {
        return dbName
    }

    fun getAppDbUrl(): String {
        return "jdbc:postgresql://${getDbHost()}:5432/$dbName?user=postgres&password=postgres"
    }

}

File: src/services/ConfigService.mjs
const DB_HOST = `127.0.0.1`;
const DB_NAME = `pgrepl`;

class ConfigService {
    get dbHost() {
        return process.env.DB_HOST || DB_HOST;
    }

    get dbName() {
        return process.env.DB_NAME || DB_NAME;
    }

    get dbUrl() {
        return `postgresql://postgres:postgres@${this.dbHost}:5432/${this.dbName}`;
    }
}

export default new ConfigService();